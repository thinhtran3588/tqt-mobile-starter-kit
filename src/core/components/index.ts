export * from './icon/icon.component';
export * from './button/button.component';
export * from './text/text.component';
export * from './text-input/text-input.component';
export * from './view/view.component';
export * from './layout/layout.component';
export * from './list-item/list-item.component';
export * from './loading/loading.component';
export * from './loading-modal/loading-modal.component';
export * from './blur/blur.component';
export * from './picker/picker.component';
export * from './picker-input/picker-input.component';
export * from './internet-connection/internet-connection.component';
export * from './confirmation/confirmation.component';
export * from './autocomplete-input/autocomplete-input.component';
export * from './autocomplete-multiple-input/autocomplete-multiple-input.component';

export {StatusBar, Alert, ScrollView, Pressable, Image} from 'react-native';

export {WebView} from 'react-native-webview';

export {TabView, SceneMap, TabBar} from 'react-native-tab-view';

export {
  Provider as PaperProvider,
  DefaultTheme,
  DarkTheme,
  Colors,
  Appbar,
  Card,
  ActivityIndicator,
  Avatar,
  Badge,
  Banner,
  Checkbox,
  List,
  Divider,
  Switch,
  IconButton,
  Menu,
} from 'react-native-paper';
