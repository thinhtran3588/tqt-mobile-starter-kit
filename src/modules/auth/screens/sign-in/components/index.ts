export * from './social-sign-in/social-sign-in.component';
