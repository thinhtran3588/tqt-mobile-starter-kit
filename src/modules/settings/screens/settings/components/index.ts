export * from './theme-setting/theme-setting.component';
export * from './primary-color-setting/primary-color-setting.component';
export * from './general-info/general-info.component';
export * from './language-setting/language-setting.component';
