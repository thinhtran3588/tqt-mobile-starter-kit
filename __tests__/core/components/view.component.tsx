import React from 'react';
import {render, toJSON} from '@test-utils';
import {Text, View} from '@core/components';

it('renders correctly', async () => {
  const {baseElement} = render(
    <View>
      <Text>Text</Text>
    </View>,
  );
  expect(toJSON(baseElement)).toMatchSnapshot();
});

it('renders row correctly', async () => {
  const {baseElement} = render(
    <View row>
      <Text>Text</Text>
    </View>,
  );
  expect(toJSON(baseElement)).toMatchSnapshot();
});
