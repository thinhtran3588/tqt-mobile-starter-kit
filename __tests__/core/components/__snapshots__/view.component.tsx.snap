// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      collapsable={false}
      pointerEvents="box-none"
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
            Object {},
            undefined,
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
                "textAlign": "left",
              },
              Array [
                Object {
                  "color": "#000000",
                },
                undefined,
                Object {
                  "fontSize": 12,
                },
              ],
            ]
          }
        >
          Text
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`renders row correctly 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      collapsable={false}
      pointerEvents="box-none"
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
              "flexWrap": "wrap",
            },
            Object {},
            undefined,
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
                "textAlign": "left",
              },
              Array [
                Object {
                  "color": "#000000",
                },
                undefined,
                Object {
                  "fontSize": 12,
                },
              ],
            ]
          }
        >
          Text
        </Text>
      </View>
    </View>
  </View>
</View>
`;
